src/nodes.coffee

```diff
@@ -3832,12 +3832,9 @@ exports.Elision = class Elision extends Base
 # it, all other loops can be manufactured. Useful in cases where you need more
 # flexibility or more speed than a comprehension can provide.
 exports.While = class While extends Base
-  constructor: (condition, options) ->
+  constructor: (@condition, {@invert, @guard, @isLoop} = {}) ->
     super()
 
-    @condition = if options?.invert then condition.invert() else condition
```

Comment:
Similar to `If`, delay inverting the condition until `compileNode()` gets called so that it's preserved for AST generation purposes